src/main/js/api/pluginManager.js

```diff
@@ -1,31 +1,37 @@
 /**
  * Provides a wrapper to interact with the plugin manager & update center
  */
-import jenkins from "../util/jenkins";
+import jenkins from "../util/jenkins"; // retained for reference, but not used now
 
-//Get plugin info (plugins + recommended plugin list) from update centers.
+// Get plugin info (plugins + recommended plugin list) from update centers.
 var plugins;
 
 var pluginManager = {};
 
+// Default 10 seconds for AJAX responses to return before triggering an error condition
+// (Note: The fetch API does not have a built-in timeout, so this option is omitted)
+var pluginManagerErrorTimeoutMillis = 10 * 1000;
+
+/**
+ * Get the initial plugin list from the update center.
+ */
 pluginManager.initialPluginList = function (handler) {
-  jenkins.get(
-    "/setupWizard/platformPluginList",
-    function (response) {
-      if (response.status !== "ok") {
-        handler.call({ isError: true, data: response.data });
-        return;
+  fetch("/setupWizard/platformPluginList")
```

Comment:
This is wrong. It neglects that Jenkins might not be running at the root of the site, e.g. when Jenkins runs under `https://<host>/jenkins/` this will fail.